### DISCUSSION TITLE
Naming clarification

### DISCUSSION THREAD

**Discussion by Nirmal4G** [dotnet/maui#41] [General]
### Clarify Naming

To me, **Maui** is like an artificial name and does not fit well with the naming of .NET libs and namespaces. When people first see `System.Maui` they don't see it as a `UI` project/library. This was the same thought that was going through my mind and some of my colleagues when I shared them the link to this project.

@coolcsh said the final product will be named as `System.UI` and I really like it. It's on point. Then the session by the Xamarin team showed `System.Maui`. So, Please clarify the naming in the roadmap or somewhere in the repo.

But if you guys reserve `System.UI` for future then at least include `UI` (_in CAPS_) in the product and the namespace.

If I were to call it, my preferred would be..

- `System.UI` (most preferred)
- `System.XUI` (Cross-UI)
- `System.MUI` (Multi-UI)

And for XAML, what is the status on XAML Standard?

**Comment by probonopd** [comment]
Also see https://github.com/dotnet/maui/issues/35.

**Comment by davidortinau** [comment]
Thanks for the post @Nirmal4G. Would you expand on what you mean by "artificial"? Would like to understand that more. 

Yes, we've been back and forth on the namespace between `System.UI` and `System.Maui`. Sorry for the confusion. I think this is a matter of some slides not getting updated. The current plan after an initial meeting with the .NET API review team is the latter. 

In terms of a XAML Standard, are you asking about changing Xamarin.Forms control names to match UWP or WPF? 

We have recently revisited the control naming with customers and developers, those with desktop experience and/or mobile. The majority feedback is for a multi-platform framework like this to not lock into a single platform naming scheme, but to unify on some broader, common standard. We have had initial conversations with the Fluent UI team and Windows team about investigating something like https://github.com/WICG/open-ui and may open a proposal around that in the future. However, at this time it's only a conversation and we are not committing to any work in this area today.

**Comment by muhaym** [comment]
@davidortinau - @Nirmal4G 's suggestion of System.UI seems elegant and the feeling is like it's something that was already there. But MaUI feels artificial (something not relevant to the System class). Also with the naming issue in the #35 thread, respectable choice would be going with anything other than Maui

**Comment by dcuccia** [comment]
+1 on System.UI

**Comment by ghost** [comment]
Definitely System.UI sound better. But for Xamarin.Forms ecosystem I prefer having something like (Microsoft.Xamarin / Microsoft.Xamarin.iOS / Microsoft.Xamarin.Android) to match frameworks style like ASP and Entityframework. Thanks.

**Comment by legistek** [comment]
> We have recently revisited the control naming with customers and developers, those with desktop experience and/or mobile. The majority feedback is for a multi-platform framework like this to not lock into a single platform naming scheme, but to unify on some broader, common standard. 

A compatibility bridge would be nice though, to help people who have a LOT invested in WPF/UWP XAML port their code more easily. There are a handful of things preventing this in XF that could  be fixed relatively easily - the [biggest](https://github.com/xamarin/Xamarin.Forms/issues/6452) being the default XML namespace having to be `http://xamarin.com/schemas/2014/forms`

**Comment by legistek** [comment]
I'd go with `System.UI` and officially call it `.NET UI` Short and sweet.

Then the KDE people will stop whining. You don't want a repeat of the "Metro/Modern/UWP/who knows what it's called anymore" fiasco.

I also think a name like ".NET UI" is a lot more professional and becoming of Microsoft. Maui sounds like a bad codename for an Apple product. No offense to anyone (I happen to love the island of Maui!)

**Comment by Nirmal4G** [comment]
> The majority feedback is for a multi-platform framework like this to not lock into a single platform naming scheme, but to unify on some broader, common standard.

By that logic, I would have `System.MaConsole`, `System.MaXml`, `System.MaNet`, `System.MaIO`... Do you see where I'm going with this... 😂

Who told you that `System.UI` corresponds to single platform? .NET (Core) 5 is cross platform now. So, It makes sense that `System.UI` is multi/cross-platform!! Follow the guidelines that Windows API team uses. Their naming is consistent with Framework guidelines. I know, technically it's not an native UI rendering on each platform but a presentation framework using the native UI APIs available on each platform. So might as well call it `System.UI.Presentation`/`System.UI.Presentors` or something. The whole of .NET primitive APIs are implemented using the target plaforms' native APIs.

**Whatever maybe, the root namespace for this repo should be `System.UI`. Unless you want to use `System.UI` in the future and this project is not really worth it's name?**

> In terms of a XAML Standard, are you asking about changing Xamarin.Forms control names to match UWP or WPF?

No, not just names but the language features. Why XAML (spec and its tools) is not open sourced yet.

**Comment by charlesroddie** [comment]
Sorry but I don't think we have struck upon anything good yet.
`Maui`: seems like a placeholder name
`System.Maui`: `System` seems unnecessary. Other UI frameworks including UWP aren't in `System`.
`System.UI`: it's not the only UI in .net. Also neither "System" nor "UI" are suitable brand names for the product.
I don't have any good ideas though.

**Comment by balivo** [comment]
Why not: System.UUI => Unified/Universal UI 🤣

**Comment by Happypig375** [comment]
`UI` can also stand for "Universal Interface". +1 vote to `System.UI`.

**Comment by balivo** [comment]
> `UI` can also stand for "Universal Interface". +1 vote to `System.UI`.

I know... UUI => ULTIMATE Universal Interface 🤣 

But... +1 vote to System.UI

**Comment by dersia** [comment]
As @davidortinau pointed out in another issue, MAUI is just an acronym, but I agree with others, that it doesn't feel good as a namespace name.
Especially if Microsoft is referring to it as MultiApplicationUserInterface-Framework (or what ever the acronym did stand for) it's not a good name for referring to it (imaging someone googling issues and will end up with MauiKit or such on stackoverflow) nor does it feel right in a namespace. 

I would also prefer `System.UI` or even better `Microsoft.UI`.
And then when referring to it I would rather go with dotnet-UI or come up with a really good new name like [Lei](https://en.m.wikipedia.org/wiki/Lei_(garland)) 😬

**Comment by schemburkar** [comment]
According to the [FAQ](https://github.com/dotnet/maui/wiki/FAQs#why-is-xamarinforms-becoming-net-maui), `Xamarin.Forms` becomes `System.Maui` and `Xamarin.Essentials` becomes `System.Devices`

For better upgrades and migration, would it be better to have alias pointing to the `Xamarin` namespaces? As well as nuget package alias to the MAUI releases? Similar to how the Office Fabric team is releasing under both `office-ui-fabric-react` and `@fluentui/react`

**Comment by GeraudFabien** [comment]
Another problem MAUI is a god in disney MOANA movie that try so much to be loved that he destroy the world. I'm not sure it's a good idea.

**Reply by balivo** [reply]
If MAUI is only related with Disney movie, ok... But MAUI have a "really darkness" history at Maori Legends...

**Comment by GeraudFabien** [comment]
I don't understand why system. It's not a namespace.
The naming of .net are in 4 catégory. 
- 3 Letters abreviation : WPF, UWP, ADO, ASP, DNX,...
- WYSIWYG 2 part name : WinForm, WebForm, XamarinForm, SignalR, EntityFramework, ASP.Net, Benchmark.Net, MonoGame, msBuild, NuGet
- WYSIWYG ; (Runtime, CLR, Extention, IOT, CLI,  ...)
- WTF : 
  - Roslyn (who is roslyn... Is it the wife of Mono.)
  - Mono (... i believe they wanted someone to make a game with it just to call it monogame [Monogamy in french])
  - Arcade (Let's play)
  - CRANK (Because client where always cranky about performance so instead of insulting them we insult our own tools)
  - Tye (the idea was great to evoke the fact that once you use it you would be tye to use it forever 🤦)
  - Orleans (... it's a very well know town...)
  - [Blazor (Hope it suits you...)](https://en.wikipedia.org/wiki/Blazer)
  - MAUI is in the WTF section. It's a divine god with dark past. A name already taken. A weird name (Multi-Application it should more multi-target user interface)(User interface isn't precise Graphical is more precise (And MAGI is only a simpson child))

PS ; I like blazor, Tye and mono. You directly know you talk about a tool even when talking with someone who don't know it.

For the name it's not hard 
- Dotnet.Form (*Form)
- .Form (WinForm)(Based on url Win.Form is windows inside of form. So why not just dotForm?)
- Desing.Net (*Form)
- UI Framework (EntityFramework)
- UI Core (I know the goal is to remove core. But core framework where fork of original on dotnet core. And when they get mature enough they lost they core part. (naming it just UI simple and stupid, pretty like all WYSIWYG solution))
- XUI (Xamarin User Interface) (It's my favorite. There is a Java and JS framework name like that but they seems not to be supported since 2014.)
- XPF (WPF)
- UDP (UWP) (Universal Design/Dotnet Plateform)
- DesignU (SignalR) (DesignUniversal)
- ...

**Reply by Happypig375** [reply]
Just call it .NET UI and use the namespace System.UI.
WPF uses the namespace System.Windows.
WinForms uses the namespace System.Windows.Forms.
UWP uses the namespace Windows.
If we are to make a cross platform UI, just make it System.UI and move on. (Currently the proposed namespace Is System.Maui.)

**Reply by Happypig375** [reply]
This namespace is also short and sweet like System.IO.
